source/CMakeLists.txt

```diff
@@ -24,6 +24,54 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
 
+set(GRAMMAR_JSON_FILES
+  ${CMAKE_CURRENT_SOURCE_DIR}/spirv.core.grammar.json
+  ${CMAKE_CURRENT_SOURCE_DIR}/extinst.glsl.std.450.grammar.json
+  ${CMAKE_CURRENT_SOURCE_DIR}/extinst.opencl.std.grammar.json
+)
+set(GRAMMAR_INC_FILES
+  ${spirv-tools_BINARY_DIR}/core.insts.inc
+  ${spirv-tools_BINARY_DIR}/glsl.std.450.insts.inc
+  ${spirv-tools_BINARY_DIR}/opencl.std.insts.inc
+)
+set(GRAMMAR_PROCESSING_SCRIPT
+  ${spirv-tools_SOURCE_DIR}/utils/generate_grammar_tables.py)
+
+add_custom_command(OUTPUT ${GRAMMAR_INC_FILES}
+  COMMAND ${PYTHON_EXECUTABLE} ${GRAMMAR_PROCESSING_SCRIPT}
+    --spirv-core-grammar=${CMAKE_CURRENT_SOURCE_DIR}/spirv.core.grammar.json
+    --extinst-glsl-grammar=${CMAKE_CURRENT_SOURCE_DIR}/extinst.glsl.std.450.grammar.json
+    --extinst-opencl-grammar=${CMAKE_CURRENT_SOURCE_DIR}/extinst.opencl.std.grammar.json
+    --core-insts-output=${spirv-tools_BINARY_DIR}/core.insts.inc
+    --glsl-insts-output=${spirv-tools_BINARY_DIR}/glsl.std.450.insts.inc
+    --opencl-insts-output=${spirv-tools_BINARY_DIR}/opencl.std.insts.inc
+    --operand-kinds-output=${spirv-tools_BINARY_DIR}/operand.kinds.inc
+  DEPENDS ${GRAMMAR_PROCESSING_SCRIPT} ${GRAMMAR_JSON_FILES}
+  COMMENT "Generate info tables for SPIR-V instructions and operands." VERBATIM)
+
+# The following .cpp files include the above generated .inc files.
+# Add those .inc files as their dependencies.
```

Comment:
Why doesn't CMake auto-detect this dependency, like it does for `.h` files included from `.cpp`??